x-pack/plugins/maps/public/util.ts

```diff
@@ -67,9 +73,21 @@ export function isRetina(): boolean {
   return window.devicePixelRatio === 2;
 }
 
-export function makePublicExecutionContext(
-  id: string,
-  description?: string
-): KibanaExecutionContext {
-  return makeExecutionContext(id, window.location.pathname, description);
+export function makePublicExecutionContext(description: string): KibanaExecutionContext {
+  const topLevelContext = getExecutionContext().get();
+  const context = makeExecutionContext({
+    url: window.location.pathname,
+    description,
+  });
+
+  // Distinguish between running in maps app vs. embedded
+  return topLevelContext.name !== context.name
+    ? {
+        ...topLevelContext,
+        child: context,
```

Comment:
I wonder if this child-parent relationship should be done somehow by the ExecutionContext service itself. I can see visualizations/discover embedded in dashboards or canvases. Not sure how we would implement them (how can they declare themselves containers of something), but it feels like this approach is pushing the problem to every plugin.

What do you think?

Reply:
This is actually a bug fix! it's up to each app to decide when it's running on it's own vs. when it's nested in another app's context. 

In most apps it is done in the embeddable layer (nest the new context within the top level one), but maps wasn't doing it (so the dashboard id wouldn't have been propagated into ES properly).